File: frontend/src/types/product.ts
// 商品信息接口
export interface Product {
  id: number
  name: string
  description?: string
  categoryId: number
  price: number
  marketPrice?: number
  costPrice?: number
  sku: string
  barcode?: string
  image?: string
  gallery?: string[]
  detail?: string
  status: number
  createdAt: string
  updatedAt: string
}

// 商品表单接口
export interface ProductForm {
  name: string
  description?: string
  categoryId: number
  price: number
  marketPrice?: number
  costPrice?: number
  sku: string
  barcode?: string
  image?: string
  gallery?: string[]
  detail?: string
  status: number
}

// 商品查询请求接口
export interface ProductQueryRequest {
  name?: string
  categoryId?: number
  status?: number
  minPrice?: number
  maxPrice?: number
  page?: number
  size?: number
  sortBy?: string
  sortOrder?: 'asc' | 'desc'
}

// 商品库存接口
export interface ProductInventory {
  id: number
  productId: number
  quantity: number
  reservedQuantity: number
  createdAt: string
  updatedAt: string
}

// 商品分类树节点接口
export interface CategoryTreeNode {
  id: number
  name: string
  children?: CategoryTreeNode[]
}

File: backend/src/main/java/com/example/ecommerce/entity/Product.java
package com.example.ecommerce.entity;

import com.example.ecommerce.enums.ProductStatus;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 商品实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Product extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /**
     * 商品名称
     */
    private String name;

    /**
     * 商品描述
     */
    private String description;

    /**
     * 分类ID
     */
    private Long categoryId;

    /**
     * 价格
     */
    private BigDecimal price;

    /**
     * 市场价
     */
    private BigDecimal marketPrice;

    /**
     * 成本价
     */
    private BigDecimal costPrice;

    /**
     * SKU
     */
    private String sku;

    /**
     * 条形码
     */
    private String barcode;

    /**
     * 商品图片
     */
    private String image;

    /**
     * 商品图集
     */
    private String gallery;

    /**
     * 商品详情
     */
    private String detail;

    /**
     * 状态
     */
    private ProductStatus status;
}